openmdao/drivers/pyoptsparse_driver.py

```diff
@@ -104,6 +107,10 @@ def __init__(self):
         self._problem = None
         self.sparsity = {}
 
+    def _setup(self, root):
+        self.supports['gradients'] = self.options['optimizer'] in grad_drivers
+        super(pyOptSparseDriver, self)._setup(root)
+
```

Comment:
Scipy optimizer also supports gradients so it should have this too. It already has a list of which internal optimizers support them.